blueocean-dashboard/src/main/js/components/InputParameters.jsx

```diff
@@ -0,0 +1,214 @@
+import React, { Component, PropTypes } from 'react';
+import isoFetch from 'isomorphic-fetch';
+import {
+  i18nTranslator,
+  ToastUtils,
+} from '@jenkins-cd/blueocean-core-js';
+import {
+  ModalView,
+  ModalBody,
+  ModalHeader,
+} from '@jenkins-cd/design-language';
+import { supportedInputTypesMapping } from './parameter/index';
+
+/**
+ * Simple helper to stop stopPropagation
+ * @param event the event we want to cancel
+ */
+const stopProp = (event) => {
+    event.stopPropagation();
+};
+
+/**
+ * Translate function
+ */
+const t = i18nTranslator('blueocean-dashboard');
+
+/**
+ * Creating a "<form/>"less form to submit the input parameters requested by the user in pipeline.
+ *
+ * We keep all form data in state and change them onChange and onToggle (depending of the parameter
+ * type). We match the different supported inputTypes with a mapping functions
+ * @see supportedInputTypesMapping
+ * That mapping delegates to the specific implementation where we further delegate to JDL components.
+ * In case you want to register a new mapping you need to edit './parameter/index' to add a new mapping
+ * and further in './parameter/commonProptypes' you need to include the new type in the oneOf array.
+ */
+export default class InputParameters extends Component {
+
+    // we start with an empty state
+    state = {};
+
+    /**
+     * react life cycle mapper to invoke the creation of the form state
+     */
+    componentWillMount() {
+        this.createFormState(this.props);
+    }
+
+    /**
+     * react life cycle mapper to invoke the creation of the form state
+     */
+    componentWillReceiveProps(nextProps) {
+        this.createFormState(nextProps);
+    }
+
+    /**
+     * Create a replica of the input parameters in state. Basically we just dump the whole item.
+     * @param props
+     */
+    createFormState(props) {
+        const { input } = props;
+        if (input) {
+            const { config = {} } = this.context;
+            const { parameters: inputParameters, _links: { self: { href } } } = input;
+            const parameters = {};
+            inputParameters.map((parameter, index) => {
+                parameters[index] = parameter;
+                return parameter;
+            });
+            this.setState({ parameters, href: `${config._rootURL}${href}/runs/` });
+        }
+    }
+
+    /**
+     * change a specific parameter value and update the state.
+     * @param index - which parameter we need to change
+     * @param event - the event leading to the change
+     */
+    changeParameter(index, event) {
+        // console.log('onChange', index, event);
+        const originalParameters = this.state.parameters;
+        originalParameters[index].defaultParameterValue.value = event;
```

Comment:
Per discussion in Gitter, it looks like the `value` property is observable here, so modifying it outside of `@action` decorator of `action()` function will throw this error when mobx `useStrict(true)` has been set. I believe that is set globally in core-js right now.

This may seem annoying but in a way I think it's good because it makes it very explicit when state change happens. Observable / data binding can be a slippery slope whenever any code can modify the data when there are shared references.

If you refer to core-js `ToastService` that's a very simple example where modification of an array is managed by two different `@action` methods. I think this is a good practice so that we're explicit and consistent of where we do state mutations. 

On the other hand, if I understand the code correctly, you are pulling metadata about the input parameters from a job, then rendering them dynamically for user modification when launching a run. To me, you don't want to modify the underlying data in the data model from this user interaction. So I think what you probably want to do is clone the data here that drives the input parameters. If you still need advantages of observability and UI data binding, I write a quick class which encapsulates the form state and is build from a clone of the pipeline data. That way this data can be modified freely without changing the underlying data model.